### ISSUE TITLE
AUTO_VEHICLE_TYPE_FETCH_FAIL: ARMv8 ArduPilot binaries

### ISSUE THREAD

**Issue by judahwo** [bluerobotics/BlueOS#2149]
### Bug description

AUTO_VEHICLE_TYPE_FETCH_FAIL
![image](https://github.com/bluerobotics/BlueOS/assets/129471786/e40e0305-7410-4b7e-8ca9-72e182c9da88)
I am on a Raspberry Pi 4B. It is not recognizing firmware that was previously installed just fine. Now it is throwing me an architecture incompatibility error
![image](https://github.com/bluerobotics/BlueOS/assets/129471786/c84ca568-5267-435b-8199-b2c27360df5f)
I basically can't do anything right now: 
![image](https://github.com/bluerobotics/BlueOS/assets/129471786/ccdf863d-066d-4b4d-9ba7-0966f731182b)
The ethernet connection is fine:
![image](https://github.com/bluerobotics/BlueOS/assets/129471786/1c79c932-ac75-417e-a4c2-1b09430c9b0e)
Internet connection is fine: 
![image](https://github.com/bluerobotics/BlueOS/assets/129471786/407e59af-52e6-4942-a89a-9d2bffcd616a)
I am new to the device and the software, and can't start testing systems until this is resolved. This is for a school senior engineering design project. 

### Steps to reproduce

Repeated error every time I enter BlueOS through browser and IP

### Primary pain point(s)

I can't download vehicle firmware, I can't access the vehicle setup. I basically can't do much of anything except test my ethernet and internet connections. I'm still learning the interface, so not having basic functionality has me at a standstill. 

### Additional context

https://discuss.bluerobotics.com/t/autopilot-manager-vehicle-type-fetch-fail/15175

### Prerequisites

- [X] I have checked to make sure that a similar request has not already been filed or fixed.

**Comment by RaulTrombin** [comment]
Hi @judahwo, could you specify the version of BlueOS you are using? You can find this information in the bottom left corner of the interface.

Additionally, could you provide the output of the uname -a command? To access the system's terminal, you can navigate to the terminal tab, and then use the "red-pill" command, which will grant you access to the system's terminal, similar to an SSH connection.

![image](https://github.com/bluerobotics/BlueOS/assets/80598030/59f63c9b-097b-48b7-a8ba-f95602f7850f)

**Comment by judahwo** [comment]
Hello, @RaulTrombin thank you for your reply.

This is the version info at the bottom left:
![image](https://github.com/bluerobotics/BlueOS/assets/129471786/b0003381-e69e-4238-9bee-84479c9e642b)

And also in the terminal:
![image](https://github.com/bluerobotics/BlueOS/assets/129471786/8dfd135f-fde8-42b8-a227-7d48aecd2279)

So, yes it is showing up as aarch64, as opposed to arm71. It's worth noting that this only started happening after the first time I accessed the 'red-pill' terminal and installed the Navigator Library (https://github.com/bluerobotics/navigator-lib). I tested some of the example functions found here: (https://docs.bluerobotics.com/navigator-lib/python/#), like the toggling of the onboard LED, PWM readings, etc. The next day I tried accessing BlueOS, and I had lost the MAVlink heartbeat and it no longer showed the firmware as installed. Basically, all the functionality that I had previously was gone and I am continuously getting the errors I described above.

**Comment by RaulTrombin** [comment]
Nice to have you using the navigator libraries! 
You can also check the Jupyter extension, fast setup to start codding with python and run tests:

![image](https://github.com/bluerobotics/BlueOS/assets/80598030/7a2e4d64-4e14-4058-a36b-0aabc545d9e8).
![image](https://github.com/bluerobotics/BlueOS/assets/80598030/d4e380d3-bd08-41c2-93d0-25bac01f6131)
I'm using it on BlueOS 1.1.1, probably it's working fine with navigator libraries as it supports both armv7 and aarch64 architectures.

But, focusing on this issue:
Maybe an "apt get upgrade" command was made? 
It will force the raspberry to upgrade the kernel, and will needs some confirmation. As we can see your system is running on the 6.1.21-v8. And somehow the raspberry newer versions are forcing to boot on aarch64, not keeping the previous architecture ( which I think is wrong...it should keep the armv7, already opened an issue for this on rpi foundation). 

You can try to fallback the kernel to be the same that comes with official BlueOS images, it should fix it.

Or fresh install and use new stable BlueOS release, which have some great improvements: 
[BlueOS 1.1.1](https://github.com/bluerobotics/BlueOS/releases/tag/1.1.1)

**Comment by judahwo** [comment]
I'm not sure about how fallback the kernel. I can try that. 

As for the fresh install, would that mean a new flash on my sd? I was using 1.1.1 stable originally:
![image](https://github.com/bluerobotics/BlueOS/assets/129471786/968d1106-a425-4c85-beea-e3c51bdd092b)

Could the issue be that the bootstrap doesn't match?:
![image](https://github.com/bluerobotics/BlueOS/assets/129471786/2d436b01-7219-4791-a3d1-0c31aaebb96c)

I see the option to update the bootstrap, but when I press it I get this warning:
![image](https://github.com/bluerobotics/BlueOS/assets/129471786/56949663-b3da-4d2b-bae6-1cc7a6f601a3)

So, I aborted that. Should I go ahead and update the bootstrap? Could that be my issue?

**Comment by RaulTrombin** [comment]
the apt get update itself isn't a problem,
but if you run apt get `upgrade` it can be, as it updates the OS's kernel, and current BlueOS releases are based on the shown previous kernel.

`sudo SKIP_WARNING=1 rpi-update 1340be4`
should downgrade to the BlueOS current supported kernel

**Comment by patrickelectric** [comment]
Hi @judahwo,

The problem that you are facing is a bit weird. Did you install BlueOS manually ? Are you using a BlueOS image that we provide ?

**Comment by judahwo** [comment]
@patrickelectric , no, I used the SD card that came with the Navigator, with the image already on it. It worked correctly initially. I think that @RaulTrombin identified that a sudo apt update may have forced it into aarch64. I just bought and received the Navigator a couple of weeks ago, so I don't really know what I'm doing yet. It's quite possible that I changed something that I shouldn't have. I'm guessing it has something to do with me accessing the 'red pill' terminal doing that update.

**Comment by judahwo** [comment]
@patrickelectric , now I don't know what's going on. I can't even pull up BlueOS with 192.168.2.2 in my browser. It's not working at all.

**Comment by patrickelectric** [comment]
Hi @judahwo, 

From what I can see on your last reply here:
https://github.com/bluerobotics/BlueOS/issues/2149#issuecomment-1745506660

It appears that you kernel does not match with what comes with BlueOS image, any upgrade or update on the kernel will result in the following problem. 

Can you flash a new SD card with BlueOS ? 
Everything should work fine, when working with Navigator libraries, in C++, Python or Rust, a kernel update should not becessary.

I would recommend you installing and using the python libraries or using our own jupyter playground.

Let me know if you need any further assistance.

**Comment by judahwo** [comment]
@patrickelectric yes, I am in class now, I was wondering if that is something I should try. I don't have a micro SD adapter with me, so I will follow the instructions in the software install guide and flash the SD card when I get home this evening. I will message here if I have any more issues. I won't do an update next time.

**Comment by patrickelectric** [comment]
Okay, sorry for this issue. We need to make it more clear for the user that a kernel upgrade is dangerous.
Other than that, be free to ask for any further assistance.

**Comment by patrickelectric** [comment]
This should be fixed in latest master
